ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
+        {
+            StringBuffer flags;
+            if (info.compareAllRows)
+                flags.append("|DFcompareall");
+            if (info.keepLeft)
+                flags.append("|DFkeepleft");
+            if (info.keepBest)
+                flags.append("|DFkeepbest");
+            if (flags.length()==0)
+                flags.append("|0");
+            doBuildUnsignedFunction(instance->startctx, "getFlags", flags.str()+1);
+        }
         if (!matchesConstantValue(info.numToKeep, 1))
+        {
             doBuildUnsignedFunction(instance->startctx, "numToKeep", info.numToKeep);
+            if (info.keepBest)
+                throwError(HQLERR_DedupBestWithKeepn);
```

Comment:
I noticed that there is no current support in the case of keepLeft=False and numToKeep>1.  I assumed this is because the performance would be adversely affected because 'numToKeep' rows will need to be cached in memory.  If the 'numToKeep' is very large, then a large number of rows retained in memory.

A similar issue would be faced if we were to implement keepBest with numToKeep>1.

Shall I create a separate ticket ticket relating to extending support for keepBest with numToKeep>1 or add a TODO comment here?